projects/plugins/jetpack/class.jetpack-post-images.php

```diff
@@ -105,6 +105,42 @@ public static function from_slideshow( $post_id, $width = 200, $height = 200 ) {
 		return $images;
 	}
 
+	/**
+	 * Filtering out images with broken URL from galleries.
+	 *
+	 * @param array $galleries Galleries.
+	 * @return array $filtered_galleries
+	 */
+	public static function filter_gallery_urls( $galleries ) {
+		$filtered_galleries = array();
+		foreach ( $galleries as $this_gallery ) {
+			if ( ! isset( $this_gallery['src'] ) ) {
+				continue;
+			}
+			$images_to_exclude = array();
+			$image_index       = 0;
+			foreach ( $this_gallery['src'] as $src_url ) {
+				if ( filter_var( $src_url, FILTER_VALIDATE_URL, FILTER_FLAG_PATH_REQUIRED ) === false ) {
+					$images_to_exclude[] = $image_index;
+				}
+				++$image_index;
+			}
+			foreach ( $images_to_exclude as $key ) {
+				unset( $this_gallery['src'][ $key ] );
+			}
+			if ( isset( $this_gallery['ids'] ) ) {
+				$ids = explode( ',', $this_gallery['ids'] );
+				foreach ( $images_to_exclude as $key ) {
+					unset( $ids[ $key ] );
+				}
+				$this_gallery['ids'] = implode( ',', $ids );
+			}
```

Comment:
This might be a bit simpler.
```suggestion
			$ids = isset( $this_gallery['ids'] ) ? explode( ',', $this_gallery['ids'] ) : array();
			// Make sure 'src' array isn't associative and has no holes.
			$this_gallery['src'] = array_values( $this_gallery['src'] );
			foreach ( $this_gallery['src'] as $idx => $src_url ) {
				if ( filter_var( $src_url, FILTER_VALIDATE_URL, FILTER_FLAG_PATH_REQUIRED ) === false ) {
					unset( $this_gallery['src'][ $idx ] );
					unset( $ids[ $idx ] );
				}
			}
			if ( isset( $this_gallery['ids'] ) ) {
				$this_gallery['ids'] = implode( ',', $ids );
			}
			// Remove any holes we introduced.
			$this_gallery['src'] = array_values( $this_gallery['src'] );
```

Reply:
Thank you for the suggestion. Looks good to me